ldes-server-admin/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/admin/spi/EventStreamConverterImpl.java

```diff
@@ -138,10 +138,11 @@ private Optional<Boolean> getBooleanResource(Model model, Property predicate) {
     }
 
     private Model getShaclFromModel(Model model) {
-        final Statement shaclStatement = model.listStatements(null, TREE_SHAPE, (Resource) null).nextStatement();
-        List<Statement> shaclStatements = retrieveAllStatements(shaclStatement.getResource(), model);
-
-        return createDefaultModel().add(shaclStatements);
+        final Model shaclModel = ModelFactory.createDefaultModel();
+        model.listStatements(null, TREE_SHAPE, (Resource) null)
+                .nextOptional()
+                .ifPresent(statement -> shaclModel.add(retrieveAllStatements(statement.getResource(), model)));
+        return shaclModel;
```

Comment:
this has the exact same behaviour as before.. what made you change this?

Reply:
If the <eventstream...> tree:shape [ TREE_SHAPE_INFORMATION ] is ommitted, then `.nextStatement` could be `null`, which could lead to a NullPointerException when trying to get the resource from that statement